lib/Sema/BuilderTransform.cpp

```diff
@@ -1714,10 +1714,17 @@ Optional<BraceStmt *> TypeChecker::applyResultBuilderBodyTransform(
   }
 
   // Solve the constraint system.
+
+  // When solving for code completion, we don't actually apply the solution to
+  // the AST but report the solutions to the callback. We also want to record
+  // fixes in the solution for better code completionr results.
+  bool forCodeCompletion = cs.getASTContext().CompletionCallback;
+
   SmallVector<Solution, 4> solutions;
-  bool solvingFailed = cs.solve(solutions);
+  bool solvingFailed = cs.solve(solutions, FreeTypeVariableBinding::Disallow,
+                                /*recordFixes=*/forCodeCompletion);
 
-  if (cs.getASTContext().CompletionCallback) {
+  if (forCodeCompletion) {
```

Comment:
One more thing - I think you might have to use `solveForCodeCompletion` here in `if (forCodeCompletion)` and keep `solve` as is after this block, it would do the right thing setting recordFixes to `true` and not filtering.